src/main/java/com/epam/lowcost/DAO/FlightDAOImpl.java

```diff
@@ -0,0 +1,151 @@
+        Long price = flight.getInitialPrice();
+        LocalDateTime depatureDate = flight.getDepartureDate();
+        LocalDateTime arrivalDate = flight.getArrivalDate();
+        Long plane_id = flight.getPlane().getId();
+
+        String sql = String.format("INSERT INTO Flights (initialPrice, plane_id, departureDate,arrivalDate,isDeleted)" +
+                        " VALUES (%d,%d,'%s','%s','%s')", price,
+                plane_id,
+                DateFormatter.format(depatureDate),
+                DateFormatter.format(arrivalDate),
+                "FALSE");
+        try (Connection conn = dataSource.getConnection();
+             Statement stmt = conn.createStatement()) {
+            int lines = stmt.executeUpdate(sql);
+            if (lines == 1) {
+                ResultSet rs = stmt.executeQuery("SELECT * FROM FLIGHTS");
+                rs.last();
+                long newId = rs.getLong("id");
+                flight.setId(newId);
+                return flight;
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+
+        }
+        flight = null;
+        return flight;
+    }
+
+    @Override
+    public Flight deleteFlight(Long id) {
+        Flight flight = getById(id);
+        flight.setDeleted(true);
+        String sql = String.format("UPDATE Flights SET isDeleted = TRUE WHERE id = '%d'", id);
+        try (Connection conn = dataSource.getConnection();
+             Statement stmt = conn.createStatement()) {
+            int lines = stmt.executeUpdate(sql);
+            if (lines == 1) {
+                return flight;
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+
+        }
+        flight = null;
+        return flight;
+
+    }
+
+    @Override
+    public Flight updateFlight(Flight flight) {
+        if (getById(flight.getId()) == null)
+            return null;
+        String sql = String.format("UPDATE Flights SET initialPrice='%d',departureDate='%s'," +
+                        "arrivalDate='%s', plane_id='%d' WHERE id = %d",
+                flight.getInitialPrice(),
+                DateFormatter.format(flight.getDepartureDate()),
+                DateFormatter.format(flight.getArrivalDate()),
+                flight.getPlane().getId(),
+                flight.getId());
+        try (Connection conn = dataSource.getConnection();
+             Statement stmt = conn.createStatement()) {
+            int lines = stmt.executeUpdate(sql);
+            if (lines == 1) {
+                return flight;
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+
+        }
+        flight = null;
+        return flight;
+    }
+
+    private Flight extractFlightFromRS(ResultSet rs) throws SQLException {
+        return Flight.builder()
+                .id(rs.getLong("id"))
+                .initialPrice(rs.getLong("initialPrice"))
+                .plane(Plane.builder().id(rs.getLong("plane_id"))
```

Comment:
Builder in Builder... Imagine that you provide review. Is it neatly ?)

Reply:
made separate method for this